packages/block-library/src/navigation/edit/index.js

```diff
@@ -283,7 +286,31 @@ function Navigation( {
 		hasResolvedCanUserCreateNavigationMenu,
 	} = useNavigationMenu( ref );
 
+	// Setup initial block insertion state.
+	// If:
+	// - not explicitly creating an **empty** menu.
+	// - there is not currently a ref to a Navigation Menu `wp_navigation`
+	// - there are some Navigation Menus
+	// - there is only a single Navigation Menu
+	//
+	// ...then automatically use that Navigation Menu by default.
+	useEffect( () => {
+		if (
+			isCreatingEmpty.current ||
+			ref ||
+			! navigationMenus?.length ||
+			navigationMenus?.length > 1
+		) {
+			return;
+		}
+
+		const firstAndOnlyNavigationMenu = navigationMenus[ 0 ];
```

Comment:
this name seems a bit obscure, I'd suggest using `mainNavigation`

Reply:
Fair enough. I've decided to revert to having to var declared. Personal preference, but if it isn't going to be explicit then it's not self documenting.